### DISCUSSION TITLE
What's the difference between `programs` and `home.packages`?

### DISCUSSION THREAD

**Discussion by jjant** [nix-community/home-manager#7461] [Q&A] [answered]
I have some programs installed through `programs`, and others through `home.packages`:

```nix
{ pkgs, ... }:
{
  # ...
  home.packages = with pkgs; [
    jq
  ];
  programs = {
    fzf.enable = true;
  };
}
```

What's the difference between the two?

**Comment by ivandimitrov8080** [comment, marked as answer]
In my experience the `packages` array provides your environment with just the executable and  `programs` provides some additional configuration and installation options by default and some that you can set as well.

**Comment by teto** [comment]
most of the time they are the same, but the executable provided by `programs` has more context thanks to your module configuration so it can be smarter and do extra-stuff like enable some user-wide configuration. When in doubt, you should enable the latter. 
Note that HM is not very consistent so `programs.XX.enable` wont necessarily add XX in PATH so you might have to double check.
